MovieViewer/MoviesSearchViewController.swift

```diff
@@ -0,0 +1,251 @@
+            MBProgressHUD.hideHUDForView(self.view, animated: true)
+            // turn off data loading flag
+            self.isMoreDataLoading = false
+            
+            guard error == nil else {
+                // reset table data and page index
+                self.movies?.removeAll()
+                self.searchMovieSettings.page = 1
+                self.tableView.reloadData()
+                // show error
+                self.showError(error!)
+                return
+            }
+            
+            if loadMore {
+                self.movies! += movies!
+                self.tableView.reloadData()
+            } else {
+                self.movies = movies
+                self.tableView.reloadData()
+                // scroll to top
+                self.tableView.setContentOffset(CGPointZero, animated: true)
+            }
+        }
+    }
+    
+    func setTheme() {
+        
+        tableView.backgroundColor = UIColor.blackColor()
+        // remove showing empty row for table incase network error
+        tableView.tableFooterView = UIView()
+        
+        searchBar.tintColor = themeColor
+        let textFieldInsideSearchBar = searchBar.valueForKey("searchField") as? UITextField
+        textFieldInsideSearchBar?.textColor = themeColor
+        searchBar.keyboardAppearance = .Dark
+        
+        errorView.backgroundColor = UIColor.whiteColor()
+        errorView.alpha = 0.8
+        errorLabel.textColor = UIColor.redColor()
+        
+        noDataLabel = UILabel(frame: CGRect(x: 0, y: 0, width: tableView.bounds.size.width, height: tableView.bounds.size.height))
+        noDataLabel.text = "no results found"
+        noDataLabel.textColor = UIColor.whiteColor()
+        noDataLabel.textAlignment = .Center
+    }
+}
+
+extension MoviesSearchViewController: UITableViewDataSource {
+    
+    // Tells the data source to return the number of rows in a given section of a table view.
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        
+        return filteredMovies?.count ?? 0
+    }
+    
+    // Asks the data source for a cell to insert in a particular location of the table view.
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        
+        let cell = tableView.dequeueReusableCellWithIdentifier("MovieCell", forIndexPath: indexPath) as! MovieCell
+        let movie = filteredMovies![indexPath.row]
+        cell.setData(movie)
+        cell.setTheme()
+        
+        return cell
+    }
+}
+
+extension MoviesSearchViewController: UITableViewDelegate {
+    
+    // Tells the delegate that the specified row is now selected.
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        
+        tableView.deselectRowAtIndexPath(indexPath, animated: true)
+    }
+}
+
+// SearchBar methods
```

Comment:
Instead of just `// SearchBar methods` a `//MARK: SearchBar Methods` is much more useful. :)